src/Arcus.Security.Providers.DockerSecrets/SecretStoreBuilderExtensions.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using Arcus.Security.Providers.DockerSecrets;
+using GuardNet;
+using Microsoft.Extensions.Configuration.KeyPerFile;
+using Microsoft.Extensions.FileProviders;
+
+// ReSharper disable once CheckNamespace
+namespace Microsoft.Extensions.Hosting
+{
+    /// <summary>
+    /// Extensions on the <see cref="SecretStoreBuilder" /> to easily provide access to Docker secrets in the secret store.
+    /// </summary>
+    public static class SecretStoreBuilderExtensions
+    {
+        /// <summary>
+        /// Adds Docker secrets (mounted as files in the Docker container) to the secret store.
+        /// </summary>
+        /// <param name="builder">The builder to add the Docker secrets provider to.</param>
+        /// <param name="directoryPath">The path inside the container where the Docker secrets are located.</param>
+        /// <param name="mutateSecretName">The optional function to mutate the secret name before looking it up.</param>
+        /// <exception cref="ArgumentNullException">Thrown when the <paramref name="builder"/> is <c>null</c></exception>
+        /// <exception cref="ArgumentException">Throw when the <paramref name="directoryPath"/> is blank</exception>
+        public static SecretStoreBuilder AddDockerSecrets(this SecretStoreBuilder builder, string directoryPath, Func<string, string> mutateSecretName = null)
```

Comment:
I've added a `mutateSecretName` argument, but I'm wondering if we should provide a default implementation here, instead of having 'null' as the default value ?

Default implementation could be `secretName => secretName.Replace(":", "__")` for instance.

Reply:
Hmm, we could discus this maybe in an issue. You could argue that you may want to set some general secret name mutation on the secret store -level instead on each provider. 🤔

Reply:
I wouldn't opt for a general mutation on secret store level, as I feel that the mutation should be specific for providers.

When you have a look at how configuration values are retrieved from the Configuration, your code typically uses `:` as the 'hierarchy separator'.  When you want to retrieve secrets, and you have multiple secret providers configured, your calling code also does not need to know which 'mutation' it needs to do.   That's why we specify this on provider level.
For instance, if I have configured the `KeyVaultProvider` and the `DockerSecretsProvider` , I want to be able to retrieve a secret by just calling `GetSecret("ConnectionStrings:MyDatabase")`.   When the `KeyVaultProvider` needs to retrieve the secret from KeyVault, the secretname needs to be changed to `ConnectionStrings--MyDatabase`.   When the secret is retrieved from the Docker secrets, the name must be changed to `ConnectionStrings__MyDatabase`.

(Maybe this translation should not even be defined via the `mutateSecret` func, but should be baked into the implementation of the `ISecretProvider` ?)

Reply:
But you're right.  We can discuss in a separate issue.  I'll open one later today.

Reply:
Started a new issue for this: https://github.com/arcus-azure/arcus.security/issues/199